### DISCUSSION TITLE
Feature request: support xmake packages (xrepo).

### DISCUSSION THREAD

**Discussion by idealvin** [microsoft/vcpkg#18498] [Ideas]
As far as I know, [xmake](https://github.com/xmake-io/xmake) is much simpler than cmake, it just lacks the support of big companies. Does Microsoft consider cooperating with the author of xmake to build a package manager based on xmake?

**Comment by nalzok** [comment]
Hi @idealvin, I think you need to clarify your intention: is this more of a question or a feature request?

**Reply by idealvin** [reply]
It's more than a feature request.

**Comment by idealvin** [comment]
Google has bazel, and perhaps Microsoft may have a similar build tool. 

xmake is a lightweight, powerful build tool, and the most important, it is really easy to learn. If Microsoft acquires [xmake](https://github.com/xmake-io/xmake) or something like that, and develops its own build tool based on xmake, lots of programmers will benefit from it.

**Comment by idealvin** [comment]
Nobody votes for this feature? ?

Life is short, I use [xmake](https://github.com/xmake-io/xmake).

**Comment by JackBoosY** [comment]
vcpkg needs to use a more stable configurator to implement the build process. Currently, I don't think vcpkg will change it.
We can add `vcpkg_configure_xmake` / `vcpkg_build_xmake` / `vcpkg_install_xmake`, but first there should be other ports that support it.

**Reply by idealvin** [reply]
> vcpkg needs to use a more stable configurator to implement the build process. Currently, I don't think vcpkg will change it.
> We can add `vcpkg_configure_xmake` / `vcpkg_build_xmake` / `vcpkg_install_xmake`, but first there should be other ports that support it.

@JackBoosY 

Hi, JackBoosY, thanks for your reply. 

Microsoft has already created many awesome tools, like the windows terminal, linux subsystem, which have improved the productivity of programmers.

xmake is another tool for productivity. Microsoft may cooperate with xmake to make it more acceptable.

**Reply by JackBoosY** [reply]
@idealvin I'm glad to see the first port use xmake.
